modules/structs/commit_status.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package structs
+
+// CommitStatusState holds the state of a Status
+// It can be "pending", "success", "error", "failure", and "warning"
+type CommitStatusState string
+
+const (
+	// CommitStatusPending is for when the Status is Pending
+	CommitStatusPending CommitStatusState = "pending"
+	// CommitStatusSuccess is for when the Status is Success
+	CommitStatusSuccess CommitStatusState = "success"
+	// CommitStatusError is for when the Status is Error
+	CommitStatusError CommitStatusState = "error"
+	// CommitStatusFailure is for when the Status is Failure
+	CommitStatusFailure CommitStatusState = "failure"
+	// CommitStatusWarning is for when the Status is Warning
+	CommitStatusWarning CommitStatusState = "warning"
+)
+
+// IsWorseThan returns true if this State is worse than the given State
+func (css CommitStatusState) IsWorseThan(css2 CommitStatusState) bool {
```

Comment:
`IsWorseOrSameAs()` maybe?

Reply:
Or NotGoodThan ? :)

Reply:
> 
> 
> Or NoBetterThan ? :)

`NoBetterThan` sounds good too. My problem was that the old name implied `<` but the function acts like `<=`.

Reply:
Hmm... It looks like there is a genuine total order here. An alternative might be:

```go
type CommitStatus int
const (
  CommitStatusPending CommitStatus = 0
  ...
)

func (c CommitStatus) String() {
   switch c {
   case CommitStatusPending:
       return "pending"
...
   }
}

```
Then just use normal operators for the state.

Reply:
@zeripath I will left this other PRs to keep this PR clear.

Reply:
@guillep2k done.